src/constraint_aware_append.c

```diff
@@ -0,0 +1,384 @@
+#include <postgres.h>
+#include <nodes/extensible.h>
+#include <nodes/plannodes.h>
+#include <parser/parsetree.h>
+#include <optimizer/plancat.h>
+#include <optimizer/clauses.h>
+#include <optimizer/prep.h>
+#include <executor/executor.h>
+#include <catalog/pg_class.h>
+#include <utils/memutils.h>
+#include <utils/lsyscache.h>
+#include <commands/explain.h>
+
+#include "constraint_aware_append.h"
+#include "hypertable.h"
+
+/*
+ * Exclude child relations (chunks) at execution time based on constraints.
+ *
+ * This functions tries to reuse as much functionality as possible from standard
+ * constraint exclusion in PostgreSQL that normally happens at planning
+ * time. Therefore, we need to fake a number of planning-related data
+ * structures.
+ *
+ * We also need to walk the expression trees of the restriction clauses and
+ * update any Vars that reference the main table to instead reference the child
+ * table (chunk) we want to exclude.
+ */
+static bool
+excluded_by_constraint(RangeTblEntry *rte, AppendRelInfo *appinfo, List *restrictinfos)
+{
+	ListCell   *lc;
+	RelOptInfo	rel = {
+		.relid = appinfo->child_relid,
+		.reloptkind = RELOPT_OTHER_MEMBER_REL,
+		.baserestrictinfo = NIL,
+	};
+	Query		parse = {
+		.resultRelation = InvalidOid,
+	};
+	PlannerGlobal glob = {
+		.boundParams = NULL,
```

Comment:
So, this is basically faking a `PlannerGlobal` because the PG code requires that you pass one. The `boundParams` is the value checked for internally (where NULL is a valid value). Now, `boundParams` would be NULL even if I don't explicitly initialize it to NULL, but thought it useful for clarity to point out that this is the parameter that the internal code cares about.